packages/metadata-service/src/digitalSpace/digitalSpace.service.ts

```diff
@@ -0,0 +1,49 @@
+import { HttpException, HttpStatus, Injectable } from '@nestjs/common';
+import * as AWS from 'aws-sdk';
+import axios from 'axios';
+import { v4 as uuidv4 } from 'uuid';
+
+@Injectable()
+export class DigitalSpaceService {
+  spacesEndpoint = new AWS.Endpoint(`${process.env['SPACE_ENDPOINT']}`);
+  s3 = new AWS.S3({
+    endpoint: `${this.spacesEndpoint.href}bookPictures`,
+    credentials: new AWS.Credentials({
+      accessKeyId: `${process.env['SPACE_ID']}`,
+      secretAccessKey: `${process.env['SPACE_SECRET']}`,
+    }),
+  });
+
+  createFile({ fileExtension, buffer }) {
+    try {
+      const fileName = uuidv4() + '.' + fileExtension;
+      return new Promise((resolve, reject) => {
+        this.s3.putObject(
+          {
+            Bucket: 'mimir-content',
+            Key: fileName,
+            Body: buffer,
+            ACL: 'public-read',
+          },
+          (error: AWS.AWSError) => {
+            if (!error) {
+              resolve(
+                `${process.env['NX_API_SPACES']}/bookPictures/${fileName}`
+              );
+            } else {
+              reject(
+                new Error(
+                  `SpacesService_ERROR: ${
+                    error.message || 'Something went wrong'
+                  }`
+                )
+              );
+            }
+          }
+        );
+      });
+    } catch (e) {
+      throw new HttpException(e.message, HttpStatus.INTERNAL_SERVER_ERROR);
+    }
+  }
+}
```

Comment:
since apiserver and metadata are two different microservices, the thing that was implemented in apiserver will not import to metadata-service, so i had to write this server here as well, just to be able to work with s3 buckets

Reply:
I meant why there are 2 similar methods in different microservices?

Reply:
because, we need to save files to space from two different microservices, sending user covers of book, and also sending covers from different websites, and this is method for holding API of digital Ocean